js/app.js

```diff
@@ -1,2 +1,179 @@
 'use strict';
+//Define how many products to show at once.
+const NUM_OF_PRODUCTS = 3;
 
+// Define how many rounds there are and the current round to increment for every product click
+const TOTAL_ROUNDS = 25;
+let round = 0;
+
+// Constructor function to create a const  = new product
+function Product(name, src) {
+  this.name = name;
+  this.src = src;
+  this.clicks = 0;
+  this.views = 0;
+}
+
+// Create Products for all included images
+const bag = new Product('bag', '/img/bag.jpg');
+const banana = new Product('banana', '/img/banana.jpg');
+const bathroom = new Product('bathroom', '/img/bathroom.jpg');
+const boots = new Product('boots', '/img/boots.jpg');
+const breakfast = new Product('breakfast', '/img/breakfast.jpg');
+const bubblegum = new Product('bubblegum', '/img/bubblegum.jpg');
+const chair = new Product('chair', '/img/chair.jpg');
+const cthulhu = new Product('cthulhu', '/img/cthulhu.jpg');
+const dogDuck = new Product('dog-duck', '/img/dog-duck.jpg');
+const dragon = new Product('dragon', '/img/dragon.jpg');
+const pen = new Product('pen', '/img/pen.jpg');
+const petSweep = new Product('pet-sweep', '/img/pet-sweep.jpg');
+const scissors = new Product('scissors', '/img/scissors.jpg');
+const shark = new Product('shark', '/img/shark.jpg');
+const sweep = new Product('sweep', '/img/sweep.png');
+const tauntaun = new Product('tauntaun', '/img/tauntaun.jpg');
+const unicorn = new Product('unicorn', '/img/unicorn.jpg');
+const waterCan = new Product('water-can', '/img/water-can.jpg');
+const wineGlass = new Product('wine-glass', '/img/wine-glass.jpg');
+
+const productsArray = [
+  bag,
+  banana,
+  bathroom,
+  boots,
+  breakfast,
+  bubblegum,
+  chair,
+  cthulhu,
+  dogDuck,
+  dragon,
+  pen,
+  petSweep,
+  scissors,
+  shark,
+  sweep,
+  tauntaun,
+  unicorn,
+  waterCan,
+  wineGlass
+];
+
+const resultsButton = document.getElementById('resultsButton');
+
+// Note: when a number of products on the page is greater or equal to half the total number of products, the requirement to have no repeats between clicks can't be met. This is handled by the boolean below
+const REPEAT_TRACK = productsArray.length / 2 >= NUM_OF_PRODUCTS ? true : false;
+
+// For incrementing clicks, define a global hash table that contains the indices for each product within the products array
+const productArrayIndices = {};
+
+// This function creates an index map of the products
+function defineProductIndices() {
+  for (let i = 0; i < productsArray.length; i++) {
+    let productName = productsArray[i].name;
+    productArrayIndices[productName] = i;
+  }
+}
+
+// To ensure that there is no repeat products from the previous render, define a global hash table that is continuously updated as the renderProducts function is called
+let currentProductArrayIndices = {};
```

Comment:
Is there a reason this is an object? Seems like it could be better as an array. Many people like to use the array.shift method here.